src/restruct_poc/step/update_system_variables.hpp

```diff
@@ -3,19 +3,21 @@
 #include <Kokkos_Core.hpp>
 #include <Kokkos_Profiling_ScopedRegion.hpp>
 
-#include "calculate_node_forces.hpp"
-#include "calculate_quadrature_point_values.hpp"
-#include "update_node_state.hpp"
+#include "assemble_inertia_matrix.hpp"
+#include "assemble_residual_vector.hpp"
+#include "assemble_stiffness_matrix.hpp"
+#include "step_parameters.hpp"
 
 #include "src/restruct_poc/beams/beams.hpp"
 #include "src/restruct_poc/beams/interpolate_to_quadrature_points.hpp"
-#include "src/restruct_poc/types.hpp"
+#include "src/restruct_poc/state/state.hpp"
+#include "src/restruct_poc/system/calculate_quadrature_point_values.hpp"
+#include "src/restruct_poc/system/calculate_tangent_operator.hpp"
+#include "src/restruct_poc/system/update_node_state.hpp"
 
 namespace openturbine {
 
-inline void UpdateState(
-    const Beams& beams, const View_Nx7& Q, const View_Nx6& V, const View_Nx6& A
-) {
+inline void UpdateSystemVariables(StepParameters& parameters, const Beams& beams, State& state) {
```

Comment:
This function is doing a lot more than updating system variables as it's name implies. I wouldn't expect it to be doing any calculations. What is considered a system variable?

Reply:
A system variable is one that is owned by the beams class.  We have three types of variables currently: system, state, and constraints.  System variables are per-element and reside in the beams class.  State variables are per-node and reside in the state class.  Constraint variables are per-constraint and reside in the constraints class.  I didn't rename beams to be system, but we could do that if you think it would help.

In truth, the function is all about computing each element's contributions to the residual, stiffness, and inertia matrices.    The others, for now at least, are just working variables.  

It also computes each node's tangent matrix, which lives in `state`.  This is inconsistent, and I can split that out into its own function.  I had several practical reasons for doing it this way, but in hindsight they are probably not good enough.  

If you look in the step function where this is called, the name `UpdateSystemVariables` provides a connection to `AssembleSystem*` and symmetry with the constraint operations (`UpdateConstraintVariables` and `AssembleConstraints*`) and then state operations.  The `Update*Variables` functions compute the individual contributions to the system, while the `Assemble*` functions assemble them into the linear system.  

I know it isn't perfect, but the idea is to make a couple steps in the right direction.  I'm open to discussion if you have some ideas.

Reply:
Thanks for the explanation on the system variables. I'm not sure that I agree with the logic on calling them variables, but I'll need to try working with it before I'll know if it should change.